cmdmod.py: runas workaround for platforms that don't set a USER env var

terminalmage opened the pull request:
Solaris doesn't set a USER env var in its default environment, that is if you run `su - someuser -c env` you get a HOME, PWD, LOGNAME, etc. env var, but not a USER. This commit makes sure that the USER env var is set to the runas user.

terminalmage committed:
cmdmod.py: runas workaround for platforms that don't set a USER env var

Solaris doesn't set a USER env var in its default environment, that is
if you run `su - someuser -c env` you get a HOME, PWD, LOGNAME, etc. env
var, but not a USER. This commit makes sure that the USER env var is set
to the runas user.

terminalmage commented:
@rallytime Let me know once this commit merges forward from 2017.7 --> oxygen, at that point I'll need to fix the run_env so we're not using unicode.

dmurphy18 reviewed (approved):
Looks good, same change could have been made for env_runas too but it works

terminalmage committed:
Move Solaris USER workaround up a bit

In its previous position, the run_env would not get the workaround when
clean_env was set to True.

dmurphy18 commented:
Looks good
